### ISSUE TITLE
How to run a command like `cat` to get the file contents inside container?

### ISSUE THREAD

**Issue by jerry-yuan** [docker-java/docker-java#1319]
I'm developing a monitor application, I need to retrieve the hardware informations of server which running the docker engine. As the result, I tried to run `docker run --rm -v /proc:/proc ubuntu cat /proc/stat` to get the CPU occupancy rate of every cores and I can successfully to get the contents of /proc/stat.
And I wrote these codes following the command which work well:
```Java
CreateContainerResponse container = dockerClient.createContainerCmd("ubuntu")
                .withAttachStdout(true)
                .withCmd("cat","/proc/stat")
                .withHostConfig(HostConfig.newHostConfig().withAutoRemove(true))
                .exec();
dockerClient.startContainerCmd(container.getId()).exec();
dockerClient.attachContainerCmd(container.getId()).withStdOut(true).exec(new AttachContainerResultCallback());
```
But I cannot get any outputs from container which should be output directly by `AttachContainerResultCallback`.
After infered the issues, I found that I can use `exec` command to cat the files from existing containers:
```Java
CreateContainerResponse container = dockerClient.createContainerCmd("ubuntu")
                .withTty(true)
                .withCmd("/bin/bash")
                .exec();
dockerClient.startContainerCmd(container.getId()).exec();
ExecCreateCmdResponse response=dockerClient.execCreateCmd(container.getId()).withCmd("cat",path).withAttachStdout(true).withAttachStderr(true).exec();

ByteArrayOutputStream stdErr=new ByteArrayOutputStream();
ByteArrayOutputStream stdOut=new ByteArrayOutputStream();

try {
        dockerClient.execStartCmd(response.getId()).exec(new ExecStartResultCallback(stdOut,stdErr)).awaitCompletion();
} catch (InterruptedException e) {
        e.printStackTrace();
        return "";
}
logger.info("stdOut:{}",stdOut.toString());
logger.info("stdErr:{}",stdErr.toString());
```
I want to know that if there is any way to retrieve file content directly instead of `exec` commands inside an existed docker?
By the way, is there any possible that I will loss some outputs if I attach to the container after starting it?
Thanks for your answering.

**Comment by bsideup** [comment]
There is `CopyFileFromContainerCmd` that you can use instead of exec-ing.

**Comment by jerry-yuan** [comment]
I'm sorry that my description is unclear. To read a ram file inside the container is one of the situations. I probably need to run `free` to get the ram usage and run `nvidia-smi` to retrieve GPUs' occupancy rates. I have tried in the first snippet and it cannot get the output either.
I want to unify the way that I retrive the hardware informations.

**Comment by jerry-yuan** [comment]
I want to run some commands in the new container that will take tiny seconds and retrieve its outputs.

**Comment by jerry-yuan** [comment]
I have found a way to run like `docker run --rm ubuntu free`.
There is some trouble in CmdExecutor implemented by Jersey. If I use Jersey to connect to the docker, the docker-java cannot attach to a container before start it like this:
```Java
                logger.info("begin to create");
                CreateContainerResponse createContainerResponse = dockerClient.createContainerCmd("ubuntu")
                        .withCmd("free", "-h")
                        .withAttachStdout(true)
                        .withHostConfig(HostConfig.newHostConfig()
                                .withAutoRemove(true))
                        .exec();
                logger.info("created");
                dockerClient.attachContainerCmd(createContainerResponse.getId())
                        .withStdOut(true)
                        .withStdErr(true)
                        .withFollowStream(true)
                        .exec(new AttachContainerResultCallback());
                dockerClient.startContainerCmd(createContainerResponse.getId()).exec();

                logger.info("executed");
```
The thread would be stuck at attachContainerCmd's exec, and switch to netty connector can fix this bug(this has been pointed out in #661 ),I don't know that if the bug is in Jersey.
